src/Microsoft.AspNet.Razor/Generator/Compiler/CodeBuilder/CSharp/CSharpCodeWriter.cs

```diff
@@ -190,18 +190,18 @@ public CSharpCodeWriter WriteReturn(string value, bool endLine)
         /// <returns>The current instance of <see cref="CSharpCodeWriter"/>.</returns>
         public CSharpCodeWriter WriteLineNumberDirective(SourceLocation location, string file)
         {
-            return WriteLineNumberDirective(location.LineIndex + 1, file);
-        }
+            if (!string.IsNullOrEmpty(location.FilePath))
```

Comment:
Just check for `null` or update the remarks for `SourceLocation`/anywhere it states that if its `null` it defaults to the currently parsed file..